internal/controller/kubeobjects/velero/requests.go

```diff
@@ -588,6 +594,91 @@ func (w objectWriter) objectDelete(o client.Object) error {
 	return nil
 }
 
+func BSLIsAvailable(bsl velero.BackupStorageLocation) bool {
+	return bsl.Status.Phase == velero.BackupStorageLocationPhaseAvailable
+}
+
+// checkBSLStatus checks if BSL is Available
+func (w objectWriter) checkBSLStatus(
+	reader client.Reader,
+	key client.ObjectKey,
+	currentBSL *velero.BackupStorageLocation,
+	bslName string,
+) (bool, error) {
+	if err := reader.Get(w.ctx, key, currentBSL); err != nil {
+		return false, err
```

Comment:
Maybe log IsNotFound explicitly so it’s clear whether the BSL never existed vs a transient read error.